torch/utils/_sympy/value_ranges.py

```diff
@@ -519,10 +519,10 @@ def mul(cls, a, b):
             return cls.and_(a, b)
 
         def safe_mul(a, b):
-            # Make unknown() * wrap(0) == wrap(0)
-            if a == 0:
+            # Make unknown() * wrap(0.0) == wrap(0.0)
+            if a == 0.0:
```

Comment:
if a is 0 (integer) does this still match it?

Reply:
Yes
```python
>>> import sympy
>>> a = sympy.Number(0)
>>> a == 0.0
True # for different sympy versions
```

Reply:
🤔

Reply:
It seems that `is_zero` is a better choice.
```python
>>> import sympy
>>> a = sympy.Number(0)
>>> a.is_zero
True
>>> b = sympy.Number(0.0)
>>> b.is_zero
True
```